camunda-7-adapter/src/test/java/org/camunda/community/migration/CamundaPlatform7AdapterTest.java

```diff
@@ -5,46 +5,118 @@
-                .addProcessModel(bpmn, "test.bpmn")
-                .send().join();
-
-        ProcessInstanceEvent processInstance = zeebeClient.newCreateInstanceCommand()
-                .bpmnProcessId("test").latestVersion()
-                .variables(Collections.singletonMap("someVariable", "value"))
-                .send().join();
-
-        waitForProcessInstanceCompleted(processInstance);
-
-        assertTrue(SampleDelegate.executed);
-        assertEquals("value", SampleDelegate.capturedVariable);
-        assertEquals("42", SampleDelegate.capturedBusinessKey);
-    }
+  @Autowired
+  private ZeebeClient zeebeClient;
+
+  @Test
+  public void testDelegateClass() {
+    BpmnModelInstance bpmn = Bpmn
+        .createExecutableProcess("test")
+        .startEvent()
+        .serviceTask()
+        .zeebeJobType("camunda-7-adapter")
+        .zeebeTaskHeader("class", "org.camunda.community.migration.SampleDelegate")
+        .endEvent()
+        .done();
+
+    zeebeClient
+        .newDeployResourceCommand()
+        .addProcessModel(bpmn, "test.bpmn")
+        .send()
+        .join();
+
+    ProcessInstanceEvent processInstance = zeebeClient
+        .newCreateInstanceCommand()
+        .bpmnProcessId("test")
+        .latestVersion()
+        .variables(Collections.singletonMap("someVariable", "value"))
+        .send()
+        .join();
+
+    waitForProcessInstanceCompleted(processInstance);
+
+    assertTrue(SampleDelegate.executed);
+    assertFalse(SampleDelegate.canReachExecutionVariable);
+    assertEquals("value", SampleDelegate.capturedVariable);
+    assertEquals("42", SampleDelegate.capturedBusinessKey);
+  }
+
+  @Test
+  public void testDelegateExpression() {
+    BpmnModelInstance bpmn = Bpmn
+        .createExecutableProcess("test")
+        .startEvent()
+        .serviceTask()
+        .zeebeJobType("camunda-7-adapter")
+        .zeebeTaskHeader("delegateExpression", "${sampleDelegate}")
+        .endEvent()
+        .done();
+
+    zeebeClient
+        .newDeployResourceCommand()
+        .addProcessModel(bpmn, "test.bpmn")
+        .send()
+        .join();
+
+    ProcessInstanceEvent processInstance = zeebeClient
+        .newCreateInstanceCommand()
+        .bpmnProcessId("test")
+        .latestVersion()
+        .variables(Collections.singletonMap("someVariable", "value"))
+        .send()
+        .join();
+
+    waitForProcessInstanceCompleted(processInstance);
+
+    assertTrue(SampleDelegate.executed);
+    assertFalse(SampleDelegate.canReachExecutionVariable);
+    assertEquals("value", SampleDelegate.capturedVariable);
+    assertEquals("42", SampleDelegate.capturedBusinessKey);
+  }
+
+  @Test
+  public void testExpression() {
+    BpmnModelInstance bpmn = Bpmn
+        .createExecutableProcess("test")
+        .startEvent()
+        .serviceTask()
+        .zeebeJobType("camunda-7-adapter")
+        .zeebeTaskHeader("expression", "${sampleDelegate.execute(execution)}")
```

Comment:
Again, a common expression implementation would make it easier to sort details later.